spec/abilities/collection_ability_spec.rb

```diff
@@ -1,164 +1,392 @@
+        it { is_expected.to be_able_to(:deposit, collection) }
+        it { is_expected.to be_able_to(:deposit, solr_document) }
+        it { is_expected.to be_able_to(:view_admin_show, collection) }
+        it { is_expected.to be_able_to(:view_admin_show, solr_document) }
+        it { is_expected.to be_able_to(:read, collection) } # edit access grants read and write
+        it { is_expected.to be_able_to(:read, solr_document) } # defined in solr_document_ability.rb
+      end
+
+      context 'for abilities NOT open to managers' do
+        it { is_expected.not_to be_able_to(:manage, Hyrax::PcdmCollection) }
+      end
     end
   end
 
   context 'when collection depositor' do
-    let!(:collection) { FactoryBot.build(:collection_lw, id: 'col_du', with_permission_template: true, collection_type: collection_type) }
-    let!(:solr_document) { SolrDocument.new(collection.to_solr) }
-
-    before do
-      create(:permission_template_access,
-             :deposit,
-             permission_template: collection.permission_template,
-             agent_type: 'user',
-             agent_id: user.user_key)
-      collection.reset_access_controls!
-    end
+    let(:current_user) { depositor }
+    let(:depositor) { create(:user, email: 'depositor@example.com') }
 
-    it 'allows deposit related abilities' do
-      is_expected.to be_able_to(:view_admin_show_any, Collection)
-      is_expected.to be_able_to(:deposit, collection)
-      is_expected.to be_able_to(:deposit, solr_document)
-      is_expected.to be_able_to(:view_admin_show, collection)
-      is_expected.to be_able_to(:view_admin_show, solr_document)
+    context 'and collection is an ActiveFedora::Base' do
+      let!(:collection) do
+        FactoryBot.build(:collection_lw, id: 'col_du',
+                                         user: user,
+                                         with_permission_template: true,
+                                         collection_type: collection_type)
+      end
+      let!(:solr_document) { SolrDocument.new(collection.to_solr) }
+
+      before do
+        create(:permission_template_access,
+               :deposit,
+               permission_template: collection.permission_template,
+               agent_type: 'user',
+               agent_id: depositor.user_key)
+        collection.reset_access_controls!
+      end
+
+      context 'for abilities open to depositor' do
+        it { is_expected.to be_able_to(:view_admin_show_any, Collection) }
+        it { is_expected.to be_able_to(:deposit, collection) }
+        it { is_expected.to be_able_to(:deposit, solr_document) }
+        it { is_expected.to be_able_to(:view_admin_show, collection) }
+        it { is_expected.to be_able_to(:view_admin_show, solr_document) }
+        it { is_expected.to be_able_to(:read, collection) }
+        it { is_expected.to be_able_to(:read, solr_document) } # defined in solr_document_ability.rb
```

Comment:
These 2 read statements were denied in the original test, but shouldn't have been.  The update to the test that revealed this error was having `user@example.com` create the collection and granting `depositor@example.com` DEPOSIT access.  Both users should have read access.

See [PermissionTemplate #read_users](https://github.com/samvera/hyrax/blob/main/app/models/hyrax/permission_template.rb#L203-L208) which sets `read_users` to be users with `view` and `deposit` access.